src/ReinforcementLearningCore/src/core/run.jl

```diff
@@ -34,10 +34,10 @@ function try_parse_kw(s)
 end
 
 struct Experiment{S}
-    policy::Any
-    env::Any
-    stop_condition::Any
-    hook::Any
+    policy::AbstractPolicy
+    env::AbstractEnv
+    stop_condition::AbstractStopCondition
+    hook::AbstractHook
```

Comment:
While we're at it, lets do this the idiosyncractic way and not use abstract-typed fields (see perfomance tips of julia doc).
```
struct Experiment{P<: AbstractPolicy,E<: ect... ,S,H}
```
where {P,E,S,H} are the type fields. I think we should also add a reset_condition field. But you don't have to do it in this PR if it's too much.

Reply:
Agree 100%, thanks for the nudge! Will add this to the PR.